docs/REST API/rest-api.md

```diff
@@ -8723,6 +8723,73 @@ Gets the result of a script that was executed.
 
 > Note: `created_at` is the creation timestamp of the script execution request.
 
+
+### Batch-run script
+
+Run a script on multiple hosts.
+
+The script will be added to each host's list of upcoming activities.
+
+`POST /api/v1/fleet/scripts/run/batch`
+
+#### Parameters
+
+| Name            | Type    | In   | Description                                                                                    |
+| ----            | ------- | ---- | --------------------------------------------                                                   |
+| script_id       | integer | body | **Required**. The ID of the existing saved script to run. |
+| host_ids        | array   | body |  List of host IDs.  Required if `filters` not specified. Only one of `host_ids` or `filters` may be included in the request.   |                                            |
+| filters | object  | body | See [filters](#filters3). Required if `host_ids` not specified. Only one of `host_ids` or `filters` may be included in the request.   |
```

Comment:
@rachaelshaw to make sure I'm understanding right:

Send a `filters` object only when "select all matching" is selected in the UI and valid filters are set

Multiple valid filters should be sent at once, so in the following scenario, run batch would be called with `query=a`, `team_id=0`, `status=online`, and `label_id=17`

![Screenshot 2025-05-07 at 4 00 57 PM](https://github.com/user-attachments/assets/5911cf80-89ab-4368-a0bc-03323694fa98)

Is that all right?

Reply:
def defer to @rachaelshaw but both these sound right to me.  Without "select all matching", we're just sending what's selected on the current page, which we can do using the host ID list.  As far as what we send, it's definitely multiple things -- whatever the "transfer" feature does in the same situation.

Reply:
@jacobshandling yes; if "select all matching" is chosen, use `filters`, otherwise use `host_ids`